### PR TITLE
Alerting: rewire expression references when queries are updated

### LINKED ISSUE #41477: Alerting: renaming queries breaks expression refs
**What happened**:

When the new Unified Alerting feature is enabled, renaming queries or server side expressions results in a bad expression evaluation.

**What you expected to happen**:

References to other queries should be re-wired when query names are updated.

**How to reproduce it (as minimally and precisely as possible)**:

1. Enable Unified Alerting
2. Create a new query and a non-empty expression that references query A
3. Rename query A

You won't be able to evaluate or test the rule as long as you do not manually update all references in the expressions.

image

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.x
- Data source type & version: N/A
- OS Grafana is installed on: N/A
- User OS & Browser: N/A
- Grafana plugins: N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by gillesdemey**
**What this PR does / why we need it**:

When renaming queries this PR will rewire the references properly for expressions.

**Which issue(s) this PR fixes**:

Fixes #41477

### CODE REVIEW

**Review by kaydelaney — COMMENTED** [review]
Nice! Maybe `queriesWithUpdatedReferences` could be factored out as a plain function and then have a test written for it to validate its behavior?

File: public/app/features/alerting/unified/components/rule-editor/QueryRows.tsx [hunk 1]
```diff
@@ -59,6 +59,47 @@ export class QueryRows extends PureComponent<Props, State> {
     );
   };
 
+  queriesWithUpdatedReferences = (queries: AlertQuery[], previousRefId: string, newRefId: string) => {
```
Comment (peterholmberg) [inline public/app/features/alerting/unified/components/rule-editor/QueryRows.tsx:62]: I feel as we should have a test for this.

**Review by peterholmberg — CHANGES_REQUESTED** [review]
Super nice, but we should have a test for the function.

**Comment by gillesdemey** [comment]
added a unit test for `queriesWithUpdatedReferences` and abstracted that into a separate function living under `utils.ts`

**Review by peterholmberg — APPROVED** [review]
Woo nice

**Pull request merged**